src/libs/antares/study/binding_constraint/BindingConstraintsRepository.cpp

```diff
@@ -73,11 +73,48 @@ static int valueForSort(BindingConstraint::Operator op)
     }
 }
 
+/**
+ * @brief Compare ranged constraint i.e their names end with _lt or _gt
+ * if suffixes are equal then the "*_lt" constraint is less than the "*_gt"
+ * @param constraint_1
+ * @param constraint_2
+ * @return true if constraint_1 < constraint_2
+ */
+bool compareRangedConstraint(const ConstraintName& constraint_1, const ConstraintName& constraint_2)
+{
+    std::string tmp_str1 = constraint_1.c_str();
+    std::string tmp_str2 = constraint_2.c_str();
+    if (auto left_pos = tmp_str1.find_last_of("_lt"); left_pos != std::string::npos)
+    {
+        std::string::iterator left = tmp_str1.begin() + left_pos;
+        size_t right_pos = 0;
+        if (right_pos = tmp_str2.find_last_of("_gt"); right_pos == std::string::npos)
+        {
+            logs.error() << "Ranged constraints comparison : " << tmp_str1 << " and " << tmp_str2
+                         << " has failed!";
+        }
+        std::string::iterator right = tmp_str2.begin() + right_pos;
+
+        auto new_s1 = std::string(tmp_str1.begin(), left - 2);
```

Comment:
Why not use use:
`auto new_s1 = tmp_str1.substr(0, left_pos);`

Reply:
thanks